contracts/src/rollup/BOLDUpgradeAction.sol

```diff
@@ -400,39 +401,15 @@ contract BOLDUpgradeAction {
         // create the config, we do this now so that we compute the expected rollup address
         Config memory config = createConfig();
 
-        // deploy the new challenge manager
-        IEdgeChallengeManager challengeManager = IEdgeChallengeManager(
-            address(
-                new TransparentUpgradeableProxy(
-                    address(IMPL_CHALLENGE_MANAGER),
-                    address(PROXY_ADMIN_BRIDGE), // use the same proxy admin as the bridge
-                    ""
-                )
-            )
-        );
-
-        // now that all the dependent contracts are pointed at the new address we can
-        // deploy and init the new rollup
-        ContractDependencies memory connectedContracts = ContractDependencies({
-            bridge: IBridge(BRIDGE),
-            sequencerInbox: ISequencerInbox(SEQ_INBOX),
-            inbox: IInbox(INBOX),
-            outbox: IOutbox(OUTBOX),
-            rollupEventInbox: IRollupEventInbox(REI),
-            challengeManager: challengeManager,
-            rollupAdminLogic: IMPL_NEW_ROLLUP_ADMIN,
-            rollupUserLogic: IRollupUser(IMPL_NEW_ROLLUP_USER),
-            validatorWalletCreator: ROLLUP_READER.validatorWalletCreator()
-        });
-
         // upgrade the surrounding contracts eg bridge, outbox, seq inbox, rollup event inbox
         // to set of the new rollup address
         bytes32 rollupSalt = keccak256(abi.encode(config));
         address expectedRollupAddress =
             Create2Upgradeable.computeAddress(rollupSalt, keccak256(type(RollupProxy).creationCode));
         upgradeSurroundingContracts(expectedRollupAddress);
 
-        challengeManager.initialize({
+        // deploy the challenge manager
+        EdgeChallengeManager challengeManager = CHALLENGE_MANAGER_FACTORY.createChallengeManager({
```

Comment:
Why do we use a factory here instead of just deploying the implementation from a template like before?

Reply:
because the `expectedRollupAddress` presumably wouldn't be known before deploying this action contract, and so we wouldn't be able to deploy the challengeManager implementation beforehand.

Reply:
it should also deploy this behind a proxy though which it doesn't right now

Reply:
[a8a0400](https://github.com/OffchainLabs/bold/pull/498/commits/a8a0400d6e32a39178eb6fdde25c8c646462309f)